src/stores/CatalogOnlineQueryProfileStore.ts

```diff
@@ -0,0 +1,222 @@
+import {action, computed, observable, makeObservable} from "mobx";
+import {CARTA} from "carta-protobuf";
+import {ControlHeader} from "stores";
+import {AbstractCatalogProfileStore, CatalogType, CatalogInfo, ProcessedColumnData} from "models";
+
+export class CatalogOnlineQueryProfileStore extends AbstractCatalogProfileStore {
+    private static readonly SimbadInitialedColumnsKeyWords = ["ra", "dec", "main_id", "coo_bibcode"];
+
+    @observable catalogInfo: CatalogInfo;
+    @observable catalogHeader: Array<CARTA.ICatalogHeader>;
+    @observable catalogControlHeader: Map<string, ControlHeader>;
+    @observable numVisibleRows: number;
+
+    constructor(catalogInfo: CatalogInfo, catalogHeader: Array<CARTA.ICatalogHeader>, catalogData: Map<number, ProcessedColumnData>, catalogType: CatalogType) {
+        super(catalogType, catalogData);
+        makeObservable(this);
+        this.catalogInfo = catalogInfo;
+        this.catalogHeader = catalogHeader.sort((a, b) => {
+            return a.columnIndex - b.columnIndex;
+        });
+        this.catalogControlHeader = this.initCatalogControlHeader;
+        this.numVisibleRows = catalogInfo.dataSize;
+
+        const coordinateSystem = catalogInfo.fileInfo.coosys[0];
+        const system = AbstractCatalogProfileStore.getCatalogSystem(coordinateSystem.system);
+        this.catalogCoordinateSystem = {
+            system: system,
+            equinox: null,
+            epoch: null,
+            coordinate: this.systemCoordinateMap.get(system)
+        };
+        this.initSortedIndexMap();
+        this.initFilterIndexMap();
+    }
+
+    @computed get initCatalogControlHeader() {
+        const controlHeaders = new Map<string, ControlHeader>();
+        const catalogHeader = this.catalogHeader;
+
+        if (catalogHeader.length) {
+            for (let index = 0; index < catalogHeader.length; index++) {
+                const header = catalogHeader[index];
+                let display = false;
+                if (CatalogOnlineQueryProfileStore.SimbadInitialedColumnsKeyWords.includes(header.name)) {
+                    display = true;
+                }
+                let controlHeader: ControlHeader = {columnIndex: header.columnIndex, dataIndex: index, display: display, filter: "", columnWidth: null};
+                controlHeaders.set(header.name, controlHeader);
+            }
+        }
+        return controlHeaders;
+    }
+
+    @computed get updateRequestDataSize() {
+        return this.catalogFilterRequest;
+    }
+
+    // do not need infinite scroll for API data
+    @computed get shouldUpdateData(): boolean {
+        return false;
+    }
+
+    @computed get loadOntoImage() {
+        return this.loadingData;
+    }
+
+    @computed get maxRows(): number {
```

Comment:
what's the point of having a `@computed` wrapper around an `@observable`? is this just so that it adheres to the parent class? If so, I'd make the `@observable` values private